### DISCUSSION TITLE
Error Messaging: Display function the user calls rather than sub-function

### DISCUSSION THREAD

**Discussion by bms63** [pharmaverse/admiral#2421] [Ideas]
For 1.2 Release, if we go forward with it.

A lot of our functions produce error messages for functions the user doesn't directly call themselves.  For this particular issue a fix was created and wondering if we should push for larger adoption.

https://github.com/pharmaverse/admiral/pull/2417#discussion_r1581429414

**Comment by bms63** [comment]
HI @ddsjoberg and @bundfussr I had a made a Discussion around the sub-function error messaging with a link to something we did in admiral.

@ddsjoberg could you provide some links to cards? and where you all are implementing your method for this "sub-function" stuff??

**Reply by ddsjoberg** [reply]
Here is the implementation (I copy this script into cards using `usethis::use_standalone()`) https://github.com/ddsjoberg/standalone/blob/main/R/standalone-cli_call_env.R

Here are general examples: https://github.com/ddsjoberg/standalone/blob/main/R/standalone-checks.R

Essentially, at the top of every user-facing function, you set the calling env with `set_cli_abort_call()`, then each time you need the env for messaging you get it with `get_cli_abort_call()`.

**Reply by bms63** [reply]
interesting!!  thanks Daniel.  FYI @pharmaverse/admiral
